### ISSUE TITLE
Naming collision

### ISSUE THREAD

**Issue by betatim** [rmsrosa/nbbinder#7]
Hey 👋,

nice work on this module and open-sourcing it! I saw it mentioned in https://discourse.jupyter.org/t/nbbinder-module-to-generate-table-of-contents-navigator-links-and-badge-links-for-a-collection-of-notebooks/3375/2

I am one of the people involved in Project Binder. Over the last three years we've been running mybinder.org which is based on https://github.com/jupyterhub/binderhub and allows people to start "binders" from their GitHub repositories.

I was wondering what Project Binder and you could do to avoid confusion about what nbbinder does and make sure people don't end up thinking it is related to mybinder.org and the "binders" people create there.

**Comment by rmsrosa** [comment]
Hello, @betatim, thanks for the comment.

Yeah, this similitude of the module names did cross my mind recently, but I was hoping it wouldn't be a big deal, since *notebookbinder* (i.e. *bookbinder* for *Jupyter notebooks*) is such a perfect name for what the module does.

But if it is, I have no problem changing the name of my module. Project Binder is well established and much more important, and it wouldn't be reasonable for that to have the name changed. I, myself, use `mybinder.og` extensively in my class, and I am very grateful to you and the other developers.

The difficult part is to come up with another good name. Do you think changing to `nbbind` or `nbbinding` would be sufficient? Or should I look into a different word derivative?

Best!

**Comment by betatim** [comment]
Finding alternatives is hard. For Binder we also have the problem that too many things are called "binder" and we've not been able to find good other names for them :-/

Some thoughts for words like bind but different: collection, staple, folder, anthology, [Diwan](https://en.wikipedia.org/wiki/Diwan_(poetry)), [Omnibus](https://en.wikipedia.org/wiki/Omnibus_edition), album, compendium, garland are some words I found that are somehow related to taking a set of existing pieces of work and combining them together.

I'll keep thinking.

**Comment by rmsrosa** [comment]
Thanks for thinking about some alternative names.

 I just thought about a good one: `nbcollate`, but it is already taken in PyPI: [nbcollate](https://pypi.org/project/nbcollate/).

I also though about `nbwrap`, but it conflicts with [nbwrapper](https://pypi.org/project/nbwrapper/).

I like your suggestion "staple", which would lead to `nbstaple` (I want the prefix `nb` to relate the module to Jupyter notebooks, of course), and I also came up with `nbgather`, `nbassemble`, and `nbpack`. All of them have one or more modules named with the corresponding name without the prefix: [staple](https://pypi.org/project/staple/), [gather](https://pypi.org/project/gather/), [assemble](https://pypi.org/project/Assemble/), and many versions of `pack`.

But all of these later modules are not related to jupyter notebooks (contrarily to mybinder), so I guess it wouldn't be too much of a problem, right? Looking at PyPI there are many similar names of packages with unrelated aims.

**Comment by betatim** [comment]
(The two hard problems in computing: cache invalidation and naming things. ... also off by one errors. 😂)

**Comment by rmsrosa** [comment]
Ok, after not thinking about this for a while, I finally decided to get over with this.

I thought about several names, but some of them had some chance of naming conflict with other packages, while others were not so good.

So I finally settled  with [nbjoint](https://github.com/rmsrosa/nbjoint). It is not as good as others, but that is it.

One of the meanings of "joint", [according to Merrian-Webster](https://www.merriam-webster.com/dictionary/joint), is, as a verb, "to unite by a joint : fit together", so there it is.

Cheers!
